File: frontend/src/services/messageService.js
import api from './api';

class MessageService {
    async getAllMessages() {
        try {
            const response = await api.get('/messages');
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch messages');
        }
    }

    async getMessageById(id) {
        try {
            const response = await api.get(`/messages/${id}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch message');
        }
    }

    async createMessage(messageData) {
        try {
            const response = await api.post('/messages', messageData);
            return response.data;
        } catch (error) {
            throw new Error('Failed to create message');
        }
    }

    async deleteMessage(id) {
        try {
            await api.delete(`/messages/${id}`);
        } catch (error) {
            throw new Error('Failed to delete message');
        }
    }

    async getMessagesBySender(senderId) {
        try {
            const response = await api.get(`/messages/sender/${senderId}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch messages by sender');
        }
    }

    async getMessagesByReceiver(receiverId) {
        try {
            const response = await api.get(`/messages/receiver/${receiverId}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch messages by receiver');
        }
    }

    async getMessagesBetweenUsers(user1Id, user2Id) {
        try {
            const response = await api.get(`/messages/conversation/${user1Id}/${user2Id}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch conversation');
        }
    }

    async getUnreadMessagesForUser(userId) {
        try {
            const response = await api.get(`/messages/unread/${userId}`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to fetch unread messages');
        }
    }

    async markAsRead(id) {
        try {
            const response = await api.put(`/messages/${id}/read`);
            return response.data;
        } catch (error) {
            throw new Error('Failed to mark message as read');
        }
    }
}

export default new MessageService(); 

File: backend/src/main/java/com/crms/message/MessageService.java
package com.crms.message;

import com.crms.user.User;
import com.crms.user.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class MessageService {

    @Autowired
    private MessageRepository messageRepository;

    @Autowired
    private UserRepository userRepository;

    public List<Message> getAllMessages() {
        return messageRepository.findAll();
    }

    public Optional<Message> getMessageById(Long id) {
        return messageRepository.findById(id);
    }

    public Message createMessage(Message message) {
        return messageRepository.save(message);
    }

    public void deleteMessage(Long id) {
        messageRepository.deleteById(id);
    }

    public List<Message> getMessagesBySender(Long senderId) {
        return messageRepository.findBySenderId(senderId);
    }

    public List<Message> getMessagesByReceiver(Long receiverId) {
        return messageRepository.findByReceiverId(receiverId);
    }

    public List<Message> getMessagesBetweenUsers(Long user1Id, Long user2Id) {
        return messageRepository.findBySenderIdAndReceiverIdOrReceiverIdAndSenderId(
                user1Id, user2Id, user1Id, user2Id);
    }

    public List<Message> getUnreadMessagesForUser(Long userId) {
        return messageRepository.findByReceiverIdAndIsReadFalse(userId);
    }

    public Message markAsRead(Long messageId) {
        Message message = messageRepository.findById(messageId)
                .orElseThrow(() -> new RuntimeException("Message not found"));
        message.setIsRead(true);
        return messageRepository.save(message);
    }
} 